book/pony/api-basics-stateless.md

```diff
@@ -4,141 +4,100 @@
 
 We'll begin digging into the Wallaroo API by creating a linear pipeline of
 stateless computations. All the code in this section can be found in
-[`examples/pony/celsius/celsius.pony`](https://github.com/WallarooLabs/wallaroo-examples/tree/{{ book.wallaroo_version }}/examples/pony/celsius/celsius.pony).
+[`examples/pony/alerts_stateless/alerts.pony`](https://github.com/WallarooLabs/wallaroo-examples/tree/{{ book.wallaroo_version }}/examples/pony/alerts_stateless/alerts.pony).
 
-We're going to start by creating an application that
-converts Celsius values to Fahrenheit. You may recall that the conversion goes
-as follows:
+We're going to start by creating an application that checks transactions to see if their amount is above or below a certain threshold (at which point we want to emit an alert). This might look something like the following:
 
 ```pony
-f = (c * 1.8) + 32
-```
-
-Even though this could define a single computation, we're going to break
-it into two to illustrate chaining. So let's start with the multiplication step:
-
-```pony
-primitive Multiply is Computation[F32, F32]
-  fun apply(input: F32): F32 =>
-    input * 1.8
+primitive CheckTransaction is StatelessComputation[Transaction, (Alert | None)]
+  fun apply(transaction: Transaction): (Alert | None) =>
+    if transaction.amount > 1000 then
+      DepositAlert(transaction.amount)
+    elseif transaction.amount < -1000 then
+      WithdrawalAlert(transaction.amount)
+    else
+      None
+    end
 
-  fun name(): String => "Multiply by 1.8"
+  fun name(): String => "CheckTransaction"
 ```
 
-A few points of interest. First, we are naming our computation `Multiply` and
-define it as a `primitive`. You can think of a `primitive` as a stateless class.
-Second, we specify that `Multiply` is a `Computation` transforming values of type
-`F32` (32-bit floats) to values of type `F32`. Third, we define an `apply` method
-that constitutes the computation logic itself. Finally, all `Computation` objects
-must implement a `name` method for performance monitoring purposes, so we include
-one here.
-
-We similarly define the addition step:
-
-```pony
-primitive Add is Computation[F32, F32]
-  fun apply(input: F32): F32 =>
-    input + 32
-
-  fun name(): String => "Add 32"
-```
+A few points of interest. First, we are naming our computation `CheckTransaction` and define it as a `primitive`. You can think of a `primitive` as a stateless class. Second, we specify that `CheckTransaction` is a `StatelessComputation` transforming values of type
```

Comment:
Should be "we are naming.. and defining" or "we will name.. and define.."

Reply:
done